timemachine/md/minimizer.py

```diff
@@ -178,38 +181,30 @@ def minimize_host_4d(mols, host_config: HostConfig, ff, mol_coords=None) -> np.n
     x0 = combined_coords
     v0 = np.zeros_like(x0)
 
-    u_impls = bind_potentials(parameterize_system(hgt, ff, 1.0))
-    x = fire_minimize(x0, u_impls, box, 50)
-
-    for lamb in np.linspace(1.0, 0, 50):
-        u_impls = bind_potentials(parameterize_system(hgt, ff, lamb))
-        # NOTE: we don't save velocities between trajectories at different lambda windows; empirically this seems to
-        # reduce the efficiency of the optimization, with more windows being required to achieve an equivalent result
-        ctxt = custom_ops.Context(x, v0, box, intg, u_impls)
+    potentials, params = parameterize_system(hgt, ff, 1.0)
+    u_impl = summed_potential_from_potentials_and_params(potentials, params)
+    bound_impls = [u_impl]
+    x = fire_minimize(x0, bound_impls, box, 50)
+
+    # No need to reconstruct the context, just change the bound potential params. Allows
+    # for preserving the velocities between windows
+    ctxt = custom_ops.Context(x, v0, box, intg, bound_impls)
+    for lamb in np.linspace(1.0, 0, 50, endpoint=False):
```

Comment:
Previously this had the endpoints included, don't think that is necessary since we minimize at each end, but could mistaken. And probably negligible compared to the speed up of using `set_params`

Reply:
Combined with the change to accumulate velocities this breaks the local MD test, though that threshold is pretty arbitrary. Put it back in as the performance didn't make a difference.